src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/clustering/SomaticClusteringModel.java

```diff
@@ -88,19 +89,33 @@ public SomaticClusteringModel(final M2FiltersArgumentCollection MTFAC, final Lis
         logClusterWeights = new double[] {Math.log1p(INITIAL_HIGH_AF_WEIGHT), Math.log(INITIAL_HIGH_AF_WEIGHT)};
     }
 
-    public void record(final int[] tumorADs, final double[] tumorLogOdds, final double artifactProbability, final double nonSomaticProbability, final VariantContext vc) {
-        // things that are definitely not somatic don't need to go in the somatic clustering model
-        if (artifactProbability > OBVIOUS_ARTIFACT_PROBABILITY_THRESHOLD) {
-            obviousArtifactCount.increment();
-            return;
-        } else if (nonSomaticProbability > OBVIOUS_ARTIFACT_PROBABILITY_THRESHOLD) {
-            return;
-        }
-
+    /**
+     * Adds data to the model for every alternate allele
+     * @param tumorADs for all alleles, summed over samples
+     * @param tumorLogOdds for alt alleles only
+     * @param artifactProbabilities by alt allele, specifically technical artifact probabilities not including sequencing error, contamination, or germline variation
+     * @param nonSomaticProbabilities by alt allele, probabilities that the variants are real but not somatic ie germline or contamination
+     * @param vc the variant context the data apply to
+     */
+    public void record(int[] tumorADs, final double[] tumorLogOdds, final List<Double> artifactProbabilities, final List<Double> nonSomaticProbabilities, final VariantContext vc) {
+        // set tumorAD to 0 for symbolic alleles so it won't contribute to overall AD
+        List<Allele> symbolicAlleles = vc.getAlternateAlleles().stream().filter(allele -> allele.isSymbolic()).collect(Collectors.toList());
+        // convert allele index to alt allele index
+        List<Integer> symIndexes = vc.getAlleleIndices(symbolicAlleles).stream().map(i -> i-1).collect(Collectors.toList());
```

Comment:
Can you go directly compute symbolic indices via `new IndexRange(0, vc.getNAlleles()).filter(n -> vc.getAllele(n).isSymbolic());`?